networkx/algorithms/euler.py

```diff
@@ -236,6 +236,9 @@ def has_eulerian_path(G):
     G : NetworkX Graph
         The graph to find an euler path in.
 
+    source : node, optional
+        Starting node for circuit.
+
```

Comment:
Maybe it'd be worth expanding on `source` in the docstring? The extended description (above the parameters section) lays out the algorithm nicely but doesn't mention `source`. I think it would be an improvement if the `source` argument could be concisely added to this description.